vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -1280,13 +1280,24 @@ public Component getListCellRendererComponent(
       // DO NOT pass value to super.getListCellRendererComponent()
       // It is incredibly inefficient for GamePieces and is not needed
       // since we overwrite the label text anyway.
+      String lineNumber;
+      if (PieceDefiner.prototypeName.isEmpty()) {
+        final int index1 = index + 1;
+        lineNumber = " ".repeat(index1 < 10 ? 2 : index1 < 100 ? 1 : 0) + index1 + ". ";
+      }
+      else {
+        lineNumber = (index == 0 ? "" : " ".repeat(index < 10 ? 2 : index < 100 ? 1 : 0) + index + ". ");
+      }
       super.getListCellRendererComponent(list, "", index, selected, hasFocus);
+
       if (value instanceof EditablePiece) {
-        setText(((EditablePiece) value).getDescription());
+        final String valueType = ((EditablePiece) value).getType();
+
+        setText((valueType.startsWith("cmt;") ? "" : lineNumber) + ((EditablePiece) value).getDescription());
```

Comment:
Try this:

        final String description = ((EditablePiece) value).getDescription();
        if (valueType.startsWith("cmt;")) {
          setText("<html>" + lineNumber.replace(" ", "&nbsp;") + description.substring(6));
        }
        else {
          setText(lineNumber + description);
        }


Actually, a better solution is to remove the HTML from Comment.getDescription (just have it return comment) and add it here in the renderer. The HTML is just for the Piece Definer, so the piece Definer should add it in. That will make this code a little simpler and things like Search that also call getDescription will not see the html.

Reply:
Great suggestion, thanks @BrentEaston. I also took advantage of the comment handling now in the renderer to alter the formatting of a comment. See what you think in the next build.